File: movie-domain/src/main/java/com/movie/domain/entity/Seat.java
package com.movie.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Getter
@NoArgsConstructor
public class Seat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String seatNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theater_id")
    private Theater theater;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        this.createdBy = "system";
        this.updatedBy = "system";
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
        this.updatedBy = "system";
    }

    public Seat(String seatNumber, Theater theater) {
        this.seatNumber = seatNumber;
        this.theater = theater;
    }
} 

File: cinema/cinema-domain/src/main/kotlin/com/study/cinema/domain/seat/Seat.kt
package com.study.cinema.domain.seat

import com.study.cinema.domain.theater.Theater
import jakarta.persistence.*

@Entity
class Seat(
    @ManyToOne(fetch = FetchType.LAZY)
    val theater: Theater,
    val positionRow: String,
    val positionColumn: String,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id = 0L
}